tests/macsec/test_neighbor_reboot.py

```diff
@@ -0,0 +1,47 @@
+'''
+
+The test case will verify the behavior when a neighbor reboots while Macsec is Active.
+
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on neighbor and reboot
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import pytest
+import logging
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology("t2")
+]
+
+
+def test_neighbor_reboot(duthosts, request, enum_frontend_dut_hostname, ctrl_links):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Neighbor type must be sonic")
+    duthost = duthosts[enum_frontend_dut_hostname]
+    if not duthost.is_macsec_capable_node():
+        pytest.skip("DUT must be a MACSec enabled device.")
+    pytest_assert(ctrl_links)
+    dut_to_neigh_int, nbr = list(ctrl_links.items())[0]
+    wait_until(120, 3, 0, lambda: duthost.iface_macsec_ok(dut_to_neigh_int) and
+               nbr["host"].iface_macsec_ok(nbr["port"]))
+    nbr["host"].shell("config save -y")
+    nbr["host"].shell("sudo reboot")
+
+    # macsec should be down on neighbor right after reboot
+    wait_until(240, 3, 0, lambda: not nbr["host"].iface_macsec_ok(nbr["port"]))
```

Comment:
Can we also check whether the MACsec of DUT is down after the reboot?

Reply:
In my testing only the neighbor's interface returned not ok with the iface_macsec_ok function.  I tried looking for other examples of checking the status before/after a reboot, but I was not able to find any.  Is there a different/better way to check I should be using?

Reply:
Maybe the reboot time of the neighbor was too short so that the DUT did not turn down its MACsec session. Since there is an indeterministic factor of timing in this case, let's do not attempt to assert that the status of MACsec of DUT is down.